pixelfusion/git-hooks
# Feature/update hooks

| Q | A |
|--|--|
| Status | on-hold |
| Migrations | NO |

## Description
Updated the scripts for `git-hooks`
- Added a README
- Updated composer file, and require all packages
- Removed the symlink and just copy/paste the files (this was giving some errors)
- Parse the hook name to the `multihook.sh` script, so the script is aware of the current hook and will fire all the code for that specific hook
- Removed the option to  parse a list with files, because we are using git (this is our source)
- Made the `git diff` output all the added `.php` files to scan
- Set a default `PHPCS_CODING_STANDARD` to `PRS2`
- Removed the FE testing, this is handled in de `front-end-stack`

## Steps to test/reproduce
1. Pull the branch local
1. To test the package in a project use the following code, in your `composer.json`
```
    "repositories": [
        {
            "type": "path",
            "url": "~/git-projects/git-hooks"
        }
    ],
    "require": {
        "pixelfusion/git-hooks": "@dev"
    },

```
1. Also add the following to your `composer.json`
```
    "scripts": {
        "post-install-cmd": [
            "./vendor/pixelfusion/git-hooks/src/git-hooks.sh"
        ],
        "post-update-cmd": [
            "./vendor/pixelfusion/git-hooks/src/git-hooks.sh"
        ]
    }
```
1. Now run `composer install` everything should setup automatically
1. Add some `.php` files to your project and make some `PSR2` mistakes
1. Add the `.php` files to `git add .` and write your commit message `git commit -m 'Testing php files'` after you commit your files the linter and PHPCS should run and give some output.
1. Check if your `PRS2` errors are fixed

## Todos
- [ ] see if we can implement phpunit and codeception
